recipes/icu/all/conanfile.py

```diff
@@ -61,15 +66,45 @@ def configure(self):
         if self.options.shared:
             del self.options.fPIC
 
+    def validate(self):
+        if not bool(self._platform):
+            raise ConanInvalidConfiguration(
+                "Compiling ICU for {} with {} not supported yet".format(str(self.settings.os),
+                                                                        str(self.settings.compiler)))
+
+    @property
+    def _platform(self):
+        return {
+            ("Windows", "Visual Studio"): "Cygwin/MSVC",
+            ("Windows", "gcc"): "MinGW",
+            ("AIX", "gcc"): "AIX/GCC",
+            ("AIX", "xlc"): "AIX",
+            ("SunOS", "gcc"): "Solaris/GCC",
+            ("Linux", "gcc"): "Linux/gcc",
+            ("Linux", "clang"): "Linux",
+            ("Macos", "gcc"): "MacOSX",
+            ("Macos", "clang"): "MacOSX",
+            ("Macos", "apple-clang"): "MacOSX",
+            ("iOS", "apple-clang"): "MacOSX",
+            ("tvOS", "apple-clang"): "MacOSX",
+            ("watchOS", "apple-clang"): "MacOSX",
+            ("FreeBSD", "gcc"): "FreeBSD",
+            ("FreeBSD", "clang"): "FreeBSD",
+        }.get((str(self.settings.os), str(self.settings.compiler)))
```

Comment:
It should be an error when platform is not available (`None`).
Perhaps test this in `validate`?
Right now, `None` is added to the argument list which fails when doing `" ".join(args)`

Reply:
It's tested in validate(). It has been added in https://github.com/conan-io/conan-center-index/pull/6652/commits/656a11d13334357211b9f0f70aaacb57f1384a1c

Reply:
Sorry, I didn't see. I was looking at the current recipe in master.